src/clients/BundleDataClient.ts

```diff
@@ -302,9 +302,11 @@ export class BundleDataClient {
       throw new Error("HubPoolClient not updated");
     }
 
-    if (blockRangesForChains.length > this.chainIdListForBundleEvaluationBlockNumbers.length) {
+    const chainIds = this.clients.configStoreClient.getChainIdIndicesForBlock(blockRangesForChains[0][0]);
+
+    if (blockRangesForChains.length > chainIds.length) {
```

Comment:
Thoughts on sanitizing this against `undefined`?
```suggestion
    if (!isDefined(chainIds) || blockRangesForChains.length > chainIds.length) {
```

Reply:
I think this check is important. At this point we should only be using chain IDs that are valid for this block range. So at this point its only programmer error if we fail this